spec/jekyll-linkpreview_spec.rb

```diff
@@ -521,55 +523,62 @@ def template_dir
     end
   end
 
-  describe '#render' do
-    before do
-      @title = 'awesome.org - an awesome organization in the world'
-      @domain = 'awesome.org'
-      @url = "https://#{@domain}/about"
-      @image = "https://#{@domain}/images/favicon.ico"
-      @description = 'An awesome organization in the world.'
-      tokenizer = Liquid::Tokenizer.new('')
-      parse_context = Liquid::ParseContext.new
-      @tag = TestLinkpreviewTag.parse(nil, @url, tokenizer, parse_context)
-
-      Dir.mkdir @tag.template_dir
-    end
+  where(:site_source) do
+    [
+      [Dir.pwd], # Default value for Jekyll
```

Comment:
I think the default value should be `nil`
We have to make sure the custom template can be rendered even when `source` is not specified.

Reply:
Getting `nil` as the `source` configuration value is actually impossible, as far as I can tell. I checked Jekyll's `Configuration` class and found this:

https://github.com/jekyll/jekyll/blob/bf67f086c4b23f2a35f1aff4244b26c81fef25c0/lib/jekyll/configuration.rb#L5-L9

This means that if `source` is missing from `_config.yml,` it defaults to `Dir.pwd`.
Also, `nil,` i.e., `source: null,` is an invalid configuration value. Even if the user deliberately specifies `null,` Jekyll still defaults to `Dir.pwd` because of the `||` operator coalescing:

https://github.com/jekyll/jekyll/blob/bf67f086c4b23f2a35f1aff4244b26c81fef25c0/lib/jekyll/configuration.rb#L100-L111

This behavior is confirmed on my side using Jekyll 4.2.2 and deliberately setting `source: null` in `_config.yml,` then writing `{{ site.source }}` in a markdown file; I still got my `Dir.pwd.`

That said, I can write that part like so:
```suggestion
      [Jekyll::Configuration::DEFAULTS["source"]],
```
I think this diff reflects our intention better; what do you think?  
This value has existed with the same class & constant structure since at least Jekyll v3.5.0, which is the minimum requirement for this plugin, so I don't expect this part to fail for older versions of Jekyll.

Reply:
> I think this diff reflects our intention better; what do you think?

Yeah, totally agree.